apps/hermes/client/js/src/HermesConnection.ts

```diff
@@ -0,0 +1,198 @@
+   * @param encoding Optional encoding type. If true, return the price update in the encoding specified by the encoding parameter. Default is hex.
+   * @param parsed Optional boolean to specify if the parsed price update should be included in the response. Default is false.
+   * @returns Array of PriceUpdate objects containing the updates at the specified timestamp.
+   */
+  async getPriceUpdatesAtTimestamp(
+    publishTime: UnixTimestamp,
+    ids: HexString[],
+    encoding?: EncodingType,
+    parsed?: boolean
+  ): Promise<PriceUpdate[]> {
+    const url = new URL(`${this.baseURL}/v2/updates/price/${publishTime}`);
+    ids.forEach((id) => url.searchParams.append("ids[]", id));
+    if (encoding) {
+      url.searchParams.append("encoding", encoding);
+    }
+    if (parsed !== undefined) {
+      url.searchParams.append("parsed", String(parsed));
+    }
+    return await this.httpRequest<PriceUpdate[]>(url.toString());
+  }
+
+  /**
+   * Fetch streaming price updates for a set of price feed IDs.
+   * This endpoint can be customized by specifying the encoding type, whether the results should include parsed updates,
+   * and if unordered updates or only benchmark updates are allowed.
+   * This will return an EventSource that can be used to listen to streaming updates.
+   * If an invalid hex-encoded ID is passed, it will throw an error.
+   *
+   *
+   * @param ids Array of hex-encoded price feed IDs for which streaming updates are requested.
+   * @param encoding Optional encoding type. If specified, updates are returned in the specified encoding. Default is hex.
+   * @param parsed Optional boolean to specify if the parsed price update should be included in the response. Default is false.
+   * @param allow_unordered Optional boolean to specify if unordered updates are allowed to be included in the stream. Default is false.
+   * @param benchmarks_only Optional boolean to specify if only benchmark prices that are the initial price updates at a given timestamp (i.e., prevPubTime != pubTime) should be returned. Default is false.
+   * @returns An EventSource instance for receiving streaming updates.
+   */
+  async getPriceUpdatesStream(
+    ids: HexString[],
+    encoding?: EncodingType,
+    parsed?: boolean,
+    allow_unordered?: boolean,
+    benchmarks_only?: boolean
+  ): Promise<EventSource> {
+    const url = new URL("/v2/updates/price/stream", this.baseURL);
+    ids.forEach((id) => {
+      url.searchParams.append("ids[]", id);
+    });
+    const params = {
+      encoding,
+      parsed: parsed !== undefined ? String(parsed) : undefined,
+      allow_unordered:
+        allow_unordered !== undefined ? String(allow_unordered) : undefined,
+      benchmarks_only:
+        benchmarks_only !== undefined ? String(benchmarks_only) : undefined,
+    };
+    Object.entries(params).forEach(([key, value]) => {
+      if (value !== undefined) {
+        url.searchParams.append(key, value);
+      }
+    });
```

Comment:
I suggest making all these optional params an `options` object argument rather than positional, i.e.:

```ts
async getPriceUpdatesStream(ids: HexString[], options?: {
  encoding?: EncodingType,
  parsed?: boolean,
  allow_unordered?: boolean,
  benchmarks_only?: boolean
}): Promise<EventSource> {
  ...
  if (options) {
    for (const [key, value] of Object.entries(options)) {
      if (value !== undefined) {
        url.searchParams.append(key, typeof value === "boolean" ? String(value) : value);
      }
    }
  }
  ...
}
```

And the same for all the other apis for the `encoding` and `parsed` params.  You can wrap up the block to append the params from the options into a nice little function too so you don't have to duplicate it to all of the methods.